tests/subsys/tracing/src/main.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+
+/**
+ * @defgroup subsys_tracing_tests subsys tracing
+ * @ingroup all_tests
+ * @{
+ * @}
+ */
+
+/* size of stack area used by each thread */
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+static struct k_thread thread;
+K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+/* define 2 semaphores */
+K_SEM_DEFINE(thread1_sem, 1, 1);
+K_SEM_DEFINE(thread2_sem, 0, 1);
+
+/* thread handle for switch */
+void thread_handle(void *p1, void *self_sem, void *other_sem)
+{
+	while (1) {
+		/* take my semaphore */
+		k_sem_take(self_sem, K_FOREVER);
+		/* wait for a while, then let other thread have a turn. */
+		k_sleep(K_MSEC(100));
+
+		k_sem_give(other_sem);
+	}
+}
+
+/**
+ * @brief Test tracing data which produced from uart backend.
+ *
+ * @details
+ * - Analyze the tracing data which produced from uart backend.
+ * - tracing data will be checked with harness function in yaml file.
+ *
+ * @ingroup subsys_tracing_tests
+ */
+void test_tracing_function_uart(void)
+{
+#ifndef CONFIG_TRACING_BACKEND_UART
```

Comment:
It would better to add CONFIG_TRACING_BACKEND_UART to filter tag in testcase.yaml.

Reply:
thanks steven for reviewing,  
we defined this config in pri_uart.conf which used in  testcase.yaml. 
I think we needn't filter this config, because I will add other cases in the main.c which need no the config.
if I filter this config, means all cases will running or skip.   and can't control a single test case.

Reply:
Got it. you are going to add more codes here so we can't move CONFIG_TRACING_BACKEND_UART to testcase.yaml.